File: cpp/1010_PairOfSongsWithTotalDurations.cpp
class Solution {
public:
    int numPairsDivisibleBy60(vector<int>& time) {
        vector<int> time_mods(60, 0);
        int count = 0;
        for (int t : time) t %= 60, time_mods[t] += 1;
        for (int t = 0; t <= 30; ++t) {
            int complement = (60 - t) % 60;
            if (t == 0 || t == 30) {
                for (int x = 2; x <= time_mods[t]; ++x) 
                    count += (x - 1);
            } else 
                count += (time_mods[t] * time_mods[complement]);
        }
        return count;
    }
};

File: python/1010_PairOfSongsWithTotalDurations.py
class Solution:
    def numPairsDivisibleBy60(self, time: List[int]) -> int:
        modded = [0 for _ in range(60)]
        count = 0
        
        for t in time: modded[t % 60] += 1
        for t in range(0, 31):
            if t == 0 or t == 30:
                for x in range(1, modded[t]):
                    count += x
            else:
                count += (modded[t] * modded[60 - t])
        
        return count
